src/circuit/gadget/sinsemilla/chip.rs

```diff
@@ -183,6 +173,10 @@ impl SinsemillaChip {
             // Y_A = (lambda_1 + lambda_2) * (x_a - x_r)
             let Y_A_next = Y_A(meta, Rotation::next());
 
+            // Check that the initial x_A, x_P, lambda_1, lambda_2 are consistent with y_Q.
+            // fixed_y_q * (2 * fixed_y_q - Y_{A,0}) = 0
+            let init_y_q_check = fixed_y_q.clone() * (two.clone() * fixed_y_q - Y_A_cur.clone());
+
             // lambda2^2 - (x_a_next + x_r + x_a_cur) = 0
             let secant_line =
                 lambda_2_cur.clone().square() - (x_a_next.clone() + x_r + x_a_cur.clone());
```

Comment:
Technically this is implementing two secant lines, because x_r is itself computed by a secant line. ("Secant" means a line between two distinct points on a curve, extended to infinity.) I think this naming is fine though.